src/env.h

```diff
@@ -92,13 +92,17 @@ struct PackageConfig {
   enum class Exists { Yes, No };
   enum class IsValid { Yes, No };
   enum class HasMain { Yes, No };
+  enum class HasExports { Yes, No };
   enum PackageType : uint32_t { None = 0, CommonJS, Module };
 
   const Exists exists;
   const IsValid is_valid;
   const HasMain has_main;
   const std::string main;
   const PackageType type;
+
+  const HasExports has_exports;
```

Comment:
Ok, one more question and I swear I’m done :smile: Is `has_exports` equivalent to `!exports.IsEmpty()`? In that case this field can be dropped/replaced by a getter, right? :)

Reply:
I think so, yes. My assumption is that @guybedford went with another field either for consistency or because he assumed the check would be more efficient. If there's no efficiency concerns, I think I'd lean towards `IsEmpty()` and drop the second field.

Reply:
@jkrems `IsEmpty()` boils down to testing a single machine word for being zero, so there is most likely no difference in performance and if there is, the benefit is on the side of using `IsEmpty()` because these objects take up less space in total and become more cacheable.